pkg/catalog/endpoint.go

```diff
@@ -38,17 +38,21 @@ func (mc *MeshCatalog) GetResolvableServiceEndpoints(svc service.MeshService) ([
 	return endpoints, nil
 }
 
-// ListAllowedEndpointsForService returns only those endpoints for a service that belong to the allowed outbound service accounts
+// ListEndpointsForServiceIdentity returns only those endpoints for a allowed outbound service accounts
 // for the given downstream identity
 // Note: ServiceIdentity must be in the format "name.namespace" [https://github.com/openservicemesh/osm/issues/3188]
-func (mc *MeshCatalog) ListAllowedEndpointsForService(downstreamIdentity identity.ServiceIdentity, upstreamSvc service.MeshService) ([]endpoint.Endpoint, error) {
+func (mc *MeshCatalog) ListEndpointsForServiceIdentity(downstreamIdentity identity.ServiceIdentity, upstreamSvc service.MeshService) ([]endpoint.Endpoint, error) {
 	outboundEndpoints, err := mc.listEndpointsForService(upstreamSvc)
 	if err != nil {
 		log.Error().Err(err).Msgf("Error looking up endpoints for upstream service %s", upstreamSvc)
 		return nil, err
 	}
+	outboundEndpointsIPSet := make(map[string]endpoint.Endpoint)
```

Comment:
Use a mapset here instead?
You could use the return value from `set.Add()` to determine if an add was successful, in which case the endpoint can be appended to the final slice.

Reply:
I didn't go with the mapset approach. The key of mapset has to be hashable, while right now endpoint.Endpoint is not.